homeassistant/components/plugwise/climate.py

```diff
@@ -122,49 +141,76 @@ def supported_features(self):
     def device_state_attributes(self):
         """Return the device specific state attributes."""
         attributes = {}
-        attributes["outdoor_temperature"] = self._outdoor_temperature
-        attributes["available_schemas"] = self._api.get_schema_names(
-            self._domain_objects
-        )
-        attributes["active_schema"] = self._active_schema
+        if self._outdoor_temperature:
+            attributes["outdoor_temperature"] = self._outdoor_temperature
+        attributes["available_schemas"] = self._schema_names
+        attributes["selected_schema"] = self._selected_schema
+        if self._boiler_temperature:
+            attributes["boiler_temperature"] = self._boiler_temperature
+        if self._water_pressure:
+            attributes["water_pressure"] = self._water_pressure
         return attributes
 
-    def update(self):
-        """Update the data from the thermostat."""
-        _LOGGER.debug("Update called")
-        self._domain_objects = self._api.get_domain_objects()
-        self._outdoor_temperature = self._api.get_outdoor_temperature(
-            self._domain_objects
-        )
-        self._active_schema = self._api.get_active_schema_name(self._domain_objects)
+    @property
+    def preset_modes(self):
+        """Return the available preset modes list and make the presets with their
```

Comment:
The first row in the docstring is a header and must fit on one line. Either unbreak this line or make the first sentence shorter. After the header line follows a  blank line and an optional body message in the docstring.

Reply:
Done